src/test/java/com/fasterxml/jackson/failing/BuilderDeserializationTest2486.java

```diff
@@ -0,0 +1,118 @@
+package com.fasterxml.jackson.failing;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.databind.BaseMapTest;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+
+import java.util.List;
+
+public class BuilderDeserializationTest2486
+        extends BaseMapTest
+{
+    @JsonDeserialize(builder = MyPOJOWithArrayCreator.Builder.class)
+    public static class MyPOJOWithArrayCreator {
+        private final int index;
+
+        private MyPOJOWithArrayCreator(int i) {
+            index = i;
+        }
+
+        public int getIndex() {
+            return index;
+        }
+
+        public static class Builder {
+            private int index;
+
+            public Builder() {
+                // Default constructor
+            }
+
+            @JsonCreator
```

Comment:
For disambiguity, should probably add `mode = Mode.DELEGATING`? (should not get confused by Jackson but still)